        if (blame->commit) {
+               column_data->flags = *(blame->commit->parent_id) ? 0 : 1;
                column_data->id = blame->commit->id;
                column_data->author = blame->commit->author;
                column_data->file_name = blame->commit->filename;
diff --git a/src/draw.c b/src/draw.c
index e5b651fed093..ae16a5477e40 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -455,6 +455,7 @@ view_column_draw(struct view *view, struct line *line, unsigned int lineno)
 {
        struct view_column *column = view->columns;
        struct view_column_data column_data = {0};
+       char *id;
 
        if (!view->ops->get_column_data(view, line, &column_data))
                return true;
@@ -485,8 +486,20 @@ view_column_draw(struct view *view, struct line *line, unsigned int lineno)
                        continue;
 
                case VIEW_COLUMN_ID:
-                       if (draw_id(view, column, column_data.reflog ? column_data.reflog : column_data.id))
+                       id = column_data.id;
+                       if (column_data.flags) {
+                               id = malloc(strlen(column_data.id) + 2);
+                               id[0] = '^';
+                               id[1] = '\0';
+                               strcat(id, column_data.id);
+                       }
+                       if (draw_id(view, column, column_data.reflog ? column_data.reflog : id)) {
+                               if (column_data.flags)
+                                       free(id);
                                return true;
+                       }
+                       if (column_data.flags)
+                               free(id);

